libbeat/tests/system/beat/beat.py

```diff
@@ -488,6 +488,21 @@ def output_has(self, lines, output_file=None):
         except IOError:
             return False
 
+    def output_is_empty(self, output_file=None):
+        """
+        Returns true if the output is empty.
+        """
+
+        # Init defaults
+        if output_file is None:
+            output_file = "output/" + self.beat_name
+
+        try:
+            with open(os.path.join(self.working_dir, output_file, ), "r", encoding="utf_8") as f:
+                return len([1 for line in f]) == 0
+        except IOError:
+            return True
```

Comment:
can't we use `self.output_has(lines=0)`?

The `output_has` should check for `None` when testing if `lines` is set, so we can correctly use `0`.

Reply:
`output_has` returns `False` if the output does not exist. I added the extra function to avoid getting an exception in this case.

Reply:
this makes me wonder if `output_has` should really raise an exception. From the test code POV we only want to know what has been 'published' (or if something has been published). The fact that the file does not exist or is empty is just a detail the test code should not care about.